sampler.go

```diff
@@ -286,33 +328,30 @@ func (s *adaptiveSampler) Close() {
 }
 
 func (s *adaptiveSampler) Equal(other Sampler) bool {
-	if o, ok := other.(*adaptiveSampler); ok {
-		if !mapKeysEqual(s.samplers, o.samplers) {
-			return false
-		}
-		if s.lowerBound != o.lowerBound || s.defaultSamplingProbability != o.defaultSamplingProbability {
-			return false
-		}
-		for operation, sampler := range s.samplers {
-			if !sampler.Equal(o.samplers[operation]) {
-				return false
-			}
-		}
-		return true
-	}
 	return false
 }
 
-func mapKeysEqual(m1, m2 map[string]Sampler) bool {
-	if len(m1) != len(m2) {
-		return false
-	}
-	for k := range m1 {
-		if _, ok := m2[k]; !ok {
-			return false
+func (s *adaptiveSampler) update(
+	lowerBound, defaultSamplingProbability float64,
+	samplingRates map[string]float64,
+) error {
+	// TODO handle case where an operation is not present in the new samplingRates, but exists in the sampler
```

Comment:
I think the simplest solution is to do nothing, leave the op in the in-memory map and wait till it goes through the calculation loop.